tests/drivers/test_drivers.py

```diff
@@ -261,3 +261,23 @@ def test_invalid_key_types(driver_path, provider):
     with pytest.raises(exceptions.InvalidKeyError) as exc:
         db.get_datasets({'not-a-key': 'val'})
     assert 'unrecognized keys' in str(exc)
+
+
+@pytest.mark.parametrize('provider', TESTABLE_DRIVERS)
+def test_use_credentials_from_settings(driver_path, provider, monkeypatch):
+    with monkeypatch.context() as m:
+        m.setenv('TC_SQL_USER', 'foo')
+        m.setenv('TC_SQL_PASSWORD', 'bar')
+
+        from terracotta import drivers, update_settings
+        update_settings()
```

Comment:
This only works because these settings have not been set explicitly by another test. We should guard against this, e.g. by clearing `_overwritten_settings` at the start of the test.

Reply:
Or shall we either amend that fixture to perform cleanup before yielding, or create a new fixture ala `ensure_default_settings`?

Reply:
Right, I forgot about that. Carry on.